File: examples/test-projects/small-projects/test-project-php/Calculator.php
<?php
/** Калькулятор — аналог Calculator из других малых проектов. */

class Calculator {
    private int $lastResult = 0;

    public function add(int $a, int $b): int {
        $result = $a + $b;
        $this->storeResult($result);
        return $result;
    }

    public function sub(int $a, int $b): int {
        $result = $a - $b;
        $this->storeResult($result);
        return $result;
    }

    public function mul(int $a, int $b): int {
        $result = $a * $b;
        $this->storeResult($result);
        return $result;
    }

    public function div(int $a, int $b): int {
        if ($b === 0) {
            return 0;
        }
        $result = intdiv($a, $b);
        $this->storeResult($result);
        return $result;
    }

    public function power(int $base, int $exp): int {
        $result = 1;
        for ($i = 0; $i < $exp; $i++) {
            $result *= $base;
        }
        $this->storeResult($result);
        return $result;
    }

    public function mod(int $a, int $b): int {
        if ($b === 0) {
            return 0;
        }
        $result = $a % $b;
        $this->storeResult($result);
        return $result;
    }

    private function storeResult(int $result): int {
        $this->lastResult = $result;
        return $this->lastResult;
    }
}

File: examples/test-projects/small-projects/test-project-python/calculator.py
"""Калькулятор — аналог Calculator из других малых проектов."""


class Calculator:
    """Простой целочисленный калькулятор с запоминанием последнего результата."""

    def __init__(self):
        self.last_result = 0

    def add(self, a, b):
        result = a + b
        self._store_result(result)
        return result

    def sub(self, a, b):
        result = a - b
        self._store_result(result)
        return result

    def mul(self, a, b):
        result = a * b
        self._store_result(result)
        return result

    def div(self, a, b):
        if b == 0:
            return 0
        result = a // b
        self._store_result(result)
        return result

    def power(self, base, exp):
        result = 1
        for _ in range(exp):
            result *= base
        self._store_result(result)
        return result

    def mod(self, a, b):
        if b == 0:
            return 0
        result = a % b
        self._store_result(result)
        return result

    def _store_result(self, result):
        self.last_result = result
        return self.last_result
